Library/Homebrew/cmd/fix-permissions.rb

```diff
@@ -0,0 +1,117 @@
+require "extend/ENV"
+
+module Homebrew
+  def fix_permissions
+    fix_permissions_shared(:wait => false)
+    system "/usr/bin/sudo", "-k"
+  end
+
+  ### !!!!!!!!!!
+  ### EVERYTHING between these markers is shared (and should be kept in sync)
+  ### with Homebrew/install/install
+  ### !!!!!!!!!!
+  def sudo *args
+    ohai "/usr/bin/sudo #{args.join ' '}"
+    system "/usr/bin/sudo", *args
+  end
+
+  def fix_permissions_shared(options={})
+    wait = options.fetch(:wait, true)
+
+    def chmod?(d)
+      File.directory?(d) && !(File.readable?(d) && File.writable?(d) && File.executable?(d))
+    end
+
+    def chown?(d)
+      File.directory?(d) && !File.owned?(d)
+    end
+
+    def chgrp?(d)
+      File.directory?(d) && !File.grpowned?(d)
+    end
+
+    def getc
+      system "/bin/stty raw -echo"
```

Comment:
Any reason to use `system "/bin/stty raw -echo"` instead of `system "/bin/stty", "raw", "-echo"`?